### ISSUE TITLE
Respawn overworld if using portal

### ISSUE THREAD

**Issue by ProgrammerDan** [Civcraft/PrisonPearl#183]
If a prisoner jumps into the end portal, they momentarily appear at 0,0 before dying of distnace damage.

**Comment by rourke750** [comment]
Are they in the overworld trying to go to the end while summoned?

**Comment by ProgrammerDan** [comment]
Ah no, this is an exit glitch. It's ignoring that the player is on the same physical server, using the "one tick respawn at main world" codepath instead of just directly respawning in the end.

**Comment by ProgrammerDan** [comment]
E..g They are pearled, and jump in the dragon's death portal, and if they press escape, momentarily appear in the overworld.

**Comment by rourke750** [comment]
https://github.com/Civcraft/PrisonPearl/blob/master/src/vg/civcraft/mc/prisonpearl/listener/PlayerListener.java#L144
That shouldnt be happening.

**Comment by ProgrammerDan** [comment]
Agreed, but I expect it's actually happening here: https://github.com/Civcraft/PrisonPearl/blob/master/src/vg/civcraft/mc/prisonpearl/listener/PlayerListener.java#L170

**Comment by rourke750** [comment]
Then they would be still send to spawn in the end https://github.com/Civcraft/PrisonPearl/blob/master/src/vg/civcraft/mc/prisonpearl/PrisonPearlUtil.java#L43

**Comment by ProgrammerDan** [comment]
No idea then, had it happen several times in testing.

**Comment by rourke750** [comment]
Can you turn debug on and see messages you get.

**Comment by ProgrammerDan** [comment]
Figured it out. It's because the summoned was still set to true. So it followed a different codepath -- e.g. the teleport was allowed, but then the distance checker killed you.

So the fix for the other issue appears to have fixed this one.

**Comment by ProgrammerDan** [comment]
Oh but new issue. If you _do_ hop in the portal, the event is cancelled, but you aren't teleported out of the end portal, so you ar ejust kind of stuck in it.

**Comment by rourke750** [comment]
What behavior do we want?

**Comment by ProgrammerDan** [comment]
If possible, effectively randomspawn them without death somewhere in the end, as if a new arrival.

**Comment by rourke750** [comment]
if they are summoned and try to go from the overworld to the end what should be done in that case?

**Comment by ProgrammerDan** [comment]
A prisoner voluntarily returning to the end? I'd assume just remove them from the summoned list, they returned themselves to jail.
